library/packages/src/modules/Package.rb

```diff
@@ -221,6 +222,22 @@ def reset
       @removed_packages.clear
     end
 
+    # Tries to find a package according to the pattern
+    #
+    # @param pattern [String] a regex pattern to match, no escaping done
+    # @return list of matching package names
+    def by_pattern(pattern)
+      raise ArgumentError, "Missing search pattern" if pattern.nil? || pattern.empty?
+
+      # NOTE: - Resolvable.find takes POSIX regexp, later select uses Ruby regexp
+      # - Resolvable.find supports regexps only for dependencies, so we need to
+      # filter result according to package name
+      Y2Packager::Resolvable.find({ provides_regexp: "^#{pattern}$" }, [:name])
+        .select { |p| p.name =~ /\A#{pattern}\z/ }
```

Comment:
I'm not sure if using the same regexp as POSIX regexp and then as a Ruby regexp is a good idea. In many cases this will work, but there are still some corner cases (like that `\A`) where it would break. In some specific case it might be OK, but I'd avoid this POSIX regexp == Ruby regexp assumption in a generic code.